From 6d6b6441774b00a6903c79d9e9d14047969d40ae Mon Sep 17 00:00:00 2001
From: Philipp Huebner <debalance@debian.org>
Date: Tue, 31 May 2022 10:30:52 +0200
Subject: [PATCH] correct includes

Many parts of ejabberd code were moved into their own projects and are
packaged separately by me. To make the build process work, several
includes must be adjusted for the corresponding libraries to be found.

Author: Philipp Huebner <debalance@debian.org>
Original: https://salsa.debian.org/ejabberd-packaging-team/ejabberd/-/blob/master/debian/patches/src.includes.patch
Signed-off-by: Johan Oudinet <johan.oudinet@gmail.com>
---
 src/ejabberd_auth.erl             | 2 +-
 src/ejabberd_auth_anonymous.erl   | 2 +-
 src/ejabberd_auth_jwt.erl         | 2 +-
 src/ejabberd_auth_mnesia.erl      | 2 +-
 src/ejabberd_auth_sql.erl         | 2 +-
 src/ejabberd_bosh.erl             | 2 +-
 src/ejabberd_c2s.erl              | 2 +-
 src/ejabberd_captcha.erl          | 2 +-
 src/ejabberd_http.erl             | 2 +-
 src/ejabberd_http_ws.erl          | 2 +-
 src/ejabberd_iq.erl               | 2 +-
 src/ejabberd_local.erl            | 2 +-
 src/ejabberd_oauth.erl            | 2 +-
 src/ejabberd_oauth_rest.erl       | 2 +-
 src/ejabberd_oauth_sql.erl        | 2 +-
 src/ejabberd_piefxis.erl          | 4 ++--
 src/ejabberd_router.erl           | 2 +-
 src/ejabberd_router_multicast.erl | 2 +-
 src/ejabberd_s2s.erl              | 2 +-
 src/ejabberd_s2s_in.erl           | 2 +-
 src/ejabberd_s2s_out.erl          | 2 +-
 src/ejabberd_service.erl          | 2 +-
 src/ejabberd_sm.erl               | 2 +-
 src/ejabberd_web.erl              | 2 +-
 src/ejabberd_web_admin.erl        | 2 +-
 src/ejabberd_websocket.erl        | 2 +-
 src/ejabberd_xmlrpc.erl           | 2 +-
 src/gen_iq_handler.erl            | 2 +-
 src/gen_pubsub_node.erl           | 2 +-
 src/gen_pubsub_nodetree.erl       | 2 +-
 src/jd2ejd.erl                    | 2 +-
 src/misc.erl                      | 2 +-
 src/mod_adhoc.erl                 | 2 +-
 src/mod_admin_extra.erl           | 2 +-
 src/mod_admin_update_sql.erl      | 2 +-
 src/mod_announce.erl              | 2 +-
 src/mod_announce_mnesia.erl       | 2 +-
 src/mod_announce_sql.erl          | 2 +-
 src/mod_avatar.erl                | 2 +-
 src/mod_block_strangers.erl       | 2 +-
 src/mod_blocking.erl              | 2 +-
 src/mod_bosh.erl                  | 2 +-
 src/mod_caps.erl                  | 2 +-
 src/mod_carboncopy.erl            | 2 +-
 src/mod_client_state.erl          | 2 +-
 src/mod_configure.erl             | 2 +-
 src/mod_conversejs.erl            | 2 +-
 src/mod_delegation.erl            | 2 +-
 src/mod_disco.erl                 | 2 +-
 src/mod_fail2ban.erl              | 2 +-
 src/mod_http_api.erl              | 2 +-
 src/mod_http_upload.erl           | 2 +-
 src/mod_http_upload_quota.erl     | 2 +-
 src/mod_jidprep.erl               | 2 +-
 src/mod_last.erl                  | 2 +-
 src/mod_legacy_auth.erl           | 2 +-
 src/mod_mam.erl                   | 2 +-
 src/mod_mam_mnesia.erl            | 2 +-
 src/mod_mam_sql.erl               | 2 +-
 src/mod_metrics.erl               | 2 +-
 src/mod_mix.erl                   | 2 +-
 src/mod_mix_pam.erl               | 2 +-
 src/mod_mqtt_session.erl          | 2 +-
 src/mod_mqtt_ws.erl               | 2 +-
 src/mod_muc.erl                   | 2 +-
 src/mod_muc_admin.erl             | 2 +-
 src/mod_muc_log.erl               | 2 +-
 src/mod_muc_mnesia.erl            | 2 +-
 src/mod_muc_room.erl              | 2 +-
 src/mod_muc_sql.erl               | 2 +-
 src/mod_multicast.erl             | 2 +-
 src/mod_offline.erl               | 2 +-
 src/mod_offline_mnesia.erl        | 2 +-
 src/mod_offline_sql.erl           | 2 +-
 src/mod_ping.erl                  | 2 +-
 src/mod_pres_counter.erl          | 2 +-
 src/mod_privacy.erl               | 2 +-
 src/mod_privacy_mnesia.erl        | 2 +-
 src/mod_privacy_sql.erl           | 2 +-
 src/mod_private.erl               | 2 +-
 src/mod_private_mnesia.erl        | 2 +-
 src/mod_private_sql.erl           | 2 +-
 src/mod_privilege.erl             | 2 +-
 src/mod_proxy65_service.erl       | 2 +-
 src/mod_pubsub.erl                | 2 +-
 src/mod_push.erl                  | 2 +-
 src/mod_push_keepalive.erl        | 2 +-
 src/mod_push_mnesia.erl           | 2 +-
 src/mod_push_sql.erl              | 2 +-
 src/mod_register.erl              | 2 +-
 src/mod_register_web.erl          | 2 +-
 src/mod_roster.erl                | 2 +-
 src/mod_roster_mnesia.erl         | 2 +-
 src/mod_roster_sql.erl            | 2 +-
 src/mod_s2s_dialback.erl          | 2 +-
 src/mod_service_log.erl           | 2 +-
 src/mod_shared_roster.erl         | 2 +-
 src/mod_shared_roster_ldap.erl    | 2 +-
 src/mod_shared_roster_mnesia.erl  | 2 +-
 src/mod_shared_roster_sql.erl     | 2 +-
 src/mod_sic.erl                   | 2 +-
 src/mod_sip.erl                   | 2 +-
 src/mod_sip_proxy.erl             | 2 +-
 src/mod_sip_registrar.erl         | 2 +-
 src/mod_stats.erl                 | 2 +-
 src/mod_stream_mgmt.erl           | 2 +-
 src/mod_stun_disco.erl            | 2 +-
 src/mod_time.erl                  | 2 +-
 src/mod_vcard.erl                 | 2 +-
 src/mod_vcard_ldap.erl            | 2 +-
 src/mod_vcard_mnesia.erl          | 2 +-
 src/mod_vcard_sql.erl             | 2 +-
 src/mod_vcard_xupdate.erl         | 2 +-
 src/mod_version.erl               | 2 +-
 src/node_flat.erl                 | 2 +-
 src/node_flat_sql.erl             | 2 +-
 src/nodetree_tree.erl             | 2 +-
 src/nodetree_tree_sql.erl         | 2 +-
 src/prosody2ejabberd.erl          | 4 ++--
 src/pubsub_subscription.erl       | 2 +-
 src/pubsub_subscription_sql.erl   | 2 +-
 test/suite.hrl                    | 8 ++++----
 tools/xml_compress_gen.erl        | 2 +-
 123 files changed, 128 insertions(+), 128 deletions(-)

diff --git a/src/ejabberd_auth.erl b/src/ejabberd_auth.erl
index 35e4921b7..7e3e05a4e 100644
--- a/src/ejabberd_auth.erl
+++ b/src/ejabberd_auth.erl
@@ -48,7 +48,7 @@
 
 -export([auth_modules/1, convert_to_scram/1]).
 
--include_lib("xmpp/include/scram.hrl").
+-include_lib("p1_xmpp/include/scram.hrl").
 -include("logger.hrl").
 
 -define(SALT_LENGTH, 16).
diff --git a/src/ejabberd_auth_anonymous.erl b/src/ejabberd_auth_anonymous.erl
index dd7ff28eb..ee56ec5b0 100644
--- a/src/ejabberd_auth_anonymous.erl
+++ b/src/ejabberd_auth_anonymous.erl
@@ -45,7 +45,7 @@
 	 plain_password_required/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/jid.hrl").
+-include_lib("p1_xmpp/include/jid.hrl").
 
 start(Host) ->
     ejabberd_hooks:add(sm_register_connection_hook, Host,
diff --git a/src/ejabberd_auth_jwt.erl b/src/ejabberd_auth_jwt.erl
index 6df742305..a8024d587 100644
--- a/src/ejabberd_auth_jwt.erl
+++ b/src/ejabberd_auth_jwt.erl
@@ -36,7 +36,7 @@
 %% 'ejabberd_hooks' callback:
 -export([check_decoded_jwt/5]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 
 %%%----------------------------------------------------------------------
diff --git a/src/ejabberd_auth_mnesia.erl b/src/ejabberd_auth_mnesia.erl
index 168ee76f5..5e75ddae2 100644
--- a/src/ejabberd_auth_mnesia.erl
+++ b/src/ejabberd_auth_mnesia.erl
@@ -37,7 +37,7 @@
 -export([need_transform/1, transform/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/scram.hrl").
+-include_lib("p1_xmpp/include/scram.hrl").
 -include("ejabberd_auth.hrl").
 
 -record(reg_users_counter, {vhost = <<"">> :: binary(),
diff --git a/src/ejabberd_auth_sql.erl b/src/ejabberd_auth_sql.erl
index 50cc1902e..29de0cd7c 100644
--- a/src/ejabberd_auth_sql.erl
+++ b/src/ejabberd_auth_sql.erl
@@ -35,7 +35,7 @@
 	 remove_user/2, store_type/1, plain_password_required/1,
 	 export/1, which_users_exists/2]).
 
--include_lib("xmpp/include/scram.hrl").
+-include_lib("p1_xmpp/include/scram.hrl").
 -include("logger.hrl").
 -include("ejabberd_sql_pt.hrl").
 -include("ejabberd_auth.hrl").
diff --git a/src/ejabberd_bosh.erl b/src/ejabberd_bosh.erl
index e9d2f0c0a..0bcc9dac0 100644
--- a/src/ejabberd_bosh.erl
+++ b/src/ejabberd_bosh.erl
@@ -43,7 +43,7 @@
 	 code_change/4]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_http.hrl").
 -include("bosh.hrl").
 
diff --git a/src/ejabberd_c2s.erl b/src/ejabberd_c2s.erl
index 3218cce51..d09f3a9b5 100644
--- a/src/ejabberd_c2s.erl
+++ b/src/ejabberd_c2s.erl
@@ -47,7 +47,7 @@
 	 reply/2, copy_state/2, set_timeout/2, route/2, format_reason/2,
 	 host_up/1, host_down/1, send_ws_ping/1, bounce_message_queue/2]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("mod_roster.hrl").
 -include("translate.hrl").
diff --git a/src/ejabberd_captcha.erl b/src/ejabberd_captcha.erl
index 343a8e54e..f3efd5026 100644
--- a/src/ejabberd_captcha.erl
+++ b/src/ejabberd_captcha.erl
@@ -42,7 +42,7 @@
 	 host_up/1, host_down/1,
 	 config_reloaded/0, process_iq/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("ejabberd_http.hrl").
 -include("translate.hrl").
diff --git a/src/ejabberd_http.erl b/src/ejabberd_http.erl
index d577a10d5..bc8e15620 100644
--- a/src/ejabberd_http.erl
+++ b/src/ejabberd_http.erl
@@ -37,7 +37,7 @@
 -export([init/3]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_http.hrl").
 -include_lib("kernel/include/file.hrl").
 
diff --git a/src/ejabberd_http_ws.erl b/src/ejabberd_http_ws.erl
index eb2100a42..632eaa5c9 100644
--- a/src/ejabberd_http_ws.erl
+++ b/src/ejabberd_http_ws.erl
@@ -36,7 +36,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -include("ejabberd_http.hrl").
 
diff --git a/src/ejabberd_iq.erl b/src/ejabberd_iq.erl
index 249c53294..2a0e1dce2 100644
--- a/src/ejabberd_iq.erl
+++ b/src/ejabberd_iq.erl
@@ -34,7 +34,7 @@
 -export([init/1, handle_call/3, handle_cast/2, handle_info/2,
 	 terminate/2, code_change/3]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("ejabberd_stacktrace.hrl").
 
diff --git a/src/ejabberd_local.erl b/src/ejabberd_local.erl
index 6dfef9d01..80f1fdbda 100644
--- a/src/ejabberd_local.erl
+++ b/src/ejabberd_local.erl
@@ -47,7 +47,7 @@
 
 -include("logger.hrl").
 -include_lib("stdlib/include/ms_transform.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_stacktrace.hrl").
 -include("translate.hrl").
 
diff --git a/src/ejabberd_oauth.erl b/src/ejabberd_oauth.erl
index 9b2c517de..0197333ef 100644
--- a/src/ejabberd_oauth.erl
+++ b/src/ejabberd_oauth.erl
@@ -54,7 +54,7 @@
          oauth_add_client_implicit/3,
          oauth_remove_client/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("ejabberd_http.hrl").
 -include("ejabberd_web_admin.hrl").
diff --git a/src/ejabberd_oauth_rest.erl b/src/ejabberd_oauth_rest.erl
index b968c9263..47d13f05e 100644
--- a/src/ejabberd_oauth_rest.erl
+++ b/src/ejabberd_oauth_rest.erl
@@ -36,7 +36,7 @@
 
 -include("ejabberd_oauth.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/jid.hrl").
+-include_lib("p1_xmpp/include/jid.hrl").
 
 init() ->
     rest:start(ejabberd_config:get_myname()),
diff --git a/src/ejabberd_oauth_sql.erl b/src/ejabberd_oauth_sql.erl
index 25be737f0..11d4483a5 100644
--- a/src/ejabberd_oauth_sql.erl
+++ b/src/ejabberd_oauth_sql.erl
@@ -37,7 +37,7 @@
 
 -include("ejabberd_oauth.hrl").
 -include("ejabberd_sql_pt.hrl").
--include_lib("xmpp/include/jid.hrl").
+-include_lib("p1_xmpp/include/jid.hrl").
 -include("logger.hrl").
 
 init() ->
diff --git a/src/ejabberd_piefxis.erl b/src/ejabberd_piefxis.erl
index d62efb300..fba76f62a 100644
--- a/src/ejabberd_piefxis.erl
+++ b/src/ejabberd_piefxis.erl
@@ -38,9 +38,9 @@
 
 -define(CHUNK_SIZE, 1024*20). %20k
 
--include_lib("xmpp/include/scram.hrl").
+-include_lib("p1_xmpp/include/scram.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_privacy.hrl").
 -include("mod_roster.hrl").
 
diff --git a/src/ejabberd_router.erl b/src/ejabberd_router.erl
index 492beb6d3..ee2ddcba9 100644
--- a/src/ejabberd_router.erl
+++ b/src/ejabberd_router.erl
@@ -69,7 +69,7 @@
 
 -include("logger.hrl").
 -include("ejabberd_router.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_stacktrace.hrl").
 
 -callback init() -> any().
diff --git a/src/ejabberd_router_multicast.erl b/src/ejabberd_router_multicast.erl
index e97ccb837..7ef026fa1 100644
--- a/src/ejabberd_router_multicast.erl
+++ b/src/ejabberd_router_multicast.erl
@@ -42,7 +42,7 @@
 	 terminate/2, code_change/3, update_to_in_wrapped/2]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -record(route_multicast, {domain = <<"">> :: binary() | '_',
 			  pid = self() :: pid()}).
diff --git a/src/ejabberd_s2s.erl b/src/ejabberd_s2s.erl
index 04490071c..0d345e947 100644
--- a/src/ejabberd_s2s.erl
+++ b/src/ejabberd_s2s.erl
@@ -52,7 +52,7 @@
 -export([get_info_s2s_connections/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_commands.hrl").
 -include_lib("stdlib/include/ms_transform.hrl").
 -include("ejabberd_stacktrace.hrl").
diff --git a/src/ejabberd_s2s_in.erl b/src/ejabberd_s2s_in.erl
index 8177a7333..fcffcb7b0 100644
--- a/src/ejabberd_s2s_in.erl
+++ b/src/ejabberd_s2s_in.erl
@@ -41,7 +41,7 @@
 -export([stop_async/1, close/1, close/2, send/2, update_state/2, establish/1,
 	 host_up/1, host_down/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 
 -type state() :: xmpp_stream_in:state().
diff --git a/src/ejabberd_s2s_out.erl b/src/ejabberd_s2s_out.erl
index f057705ed..4139b0fe7 100644
--- a/src/ejabberd_s2s_out.erl
+++ b/src/ejabberd_s2s_out.erl
@@ -39,7 +39,7 @@
 -export([start/3, start_link/3, connect/1, close/1, close/2, stop_async/1, send/2,
 	 route/2, establish/1, update_state/2, host_up/1, host_down/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
 
diff --git a/src/ejabberd_service.erl b/src/ejabberd_service.erl
index b854bcb2b..3cc939679 100644
--- a/src/ejabberd_service.erl
+++ b/src/ejabberd_service.erl
@@ -35,7 +35,7 @@
 %% API
 -export([send/2, close/1, close/2, stop_async/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
 
diff --git a/src/ejabberd_sm.erl b/src/ejabberd_sm.erl
index afb95c10b..e5011ed23 100644
--- a/src/ejabberd_sm.erl
+++ b/src/ejabberd_sm.erl
@@ -85,7 +85,7 @@
 	 handle_info/2, terminate/2, code_change/3]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_commands.hrl").
 -include("ejabberd_sm.hrl").
 -include("ejabberd_stacktrace.hrl").
diff --git a/src/ejabberd_web.erl b/src/ejabberd_web.erl
index 5cfbb93aa..8a6b0fcd4 100644
--- a/src/ejabberd_web.erl
+++ b/src/ejabberd_web.erl
@@ -33,7 +33,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -include("ejabberd_http.hrl").
 
diff --git a/src/ejabberd_web_admin.erl b/src/ejabberd_web_admin.erl
index ee50a0031..f97357f2f 100644
--- a/src/ejabberd_web_admin.erl
+++ b/src/ejabberd_web_admin.erl
@@ -35,7 +35,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -include("ejabberd_http.hrl").
 
diff --git a/src/ejabberd_websocket.erl b/src/ejabberd_websocket.erl
index 5f9d284a6..db7c53768 100644
--- a/src/ejabberd_websocket.erl
+++ b/src/ejabberd_websocket.erl
@@ -45,7 +45,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -include("ejabberd_http.hrl").
 
diff --git a/src/ejabberd_xmlrpc.erl b/src/ejabberd_xmlrpc.erl
index 1a1ec63f4..a76216963 100644
--- a/src/ejabberd_xmlrpc.erl
+++ b/src/ejabberd_xmlrpc.erl
@@ -39,7 +39,7 @@
 -include("ejabberd_http.hrl").
 -include("mod_roster.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -record(state,
 	{auth = noauth        :: noauth | map(),
diff --git a/src/gen_iq_handler.erl b/src/gen_iq_handler.erl
index 29057435d..4fee8caf7 100644
--- a/src/gen_iq_handler.erl
+++ b/src/gen_iq_handler.erl
@@ -35,7 +35,7 @@
 -deprecated([{add_iq_handler, 6}, {handle, 5}, {iqdisc, 1}]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 -include("ejabberd_stacktrace.hrl").
 
diff --git a/src/gen_pubsub_node.erl b/src/gen_pubsub_node.erl
index 3f83fe48f..5545d51de 100644
--- a/src/gen_pubsub_node.erl
+++ b/src/gen_pubsub_node.erl
@@ -25,7 +25,7 @@
 
 -module(gen_pubsub_node).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -type(host() :: mod_pubsub:host()).
 -type(nodeId() :: mod_pubsub:nodeId()).
diff --git a/src/gen_pubsub_nodetree.erl b/src/gen_pubsub_nodetree.erl
index b6b73b8cb..8c2a2a4fa 100644
--- a/src/gen_pubsub_nodetree.erl
+++ b/src/gen_pubsub_nodetree.erl
@@ -36,7 +36,7 @@
 	ServerHost :: binary(),
 	Opts :: [any()]) -> atom().
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -callback terminate(Host :: host(), ServerHost :: binary()) -> atom().
 
diff --git a/src/jd2ejd.erl b/src/jd2ejd.erl
index 279dd6e0d..ec0b2f083 100644
--- a/src/jd2ejd.erl
+++ b/src/jd2ejd.erl
@@ -31,7 +31,7 @@
 -export([import_file/1, import_dir/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 %%%----------------------------------------------------------------------
 %%% API
diff --git a/src/misc.erl b/src/misc.erl
index 474fad615..489052bd1 100644
--- a/src/misc.erl
+++ b/src/misc.erl
@@ -50,7 +50,7 @@
 	     {encode_base64, 1}]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include_lib("kernel/include/file.hrl").
 
 -type distance_cache() :: #{{string(), string()} => non_neg_integer()}.
diff --git a/src/mod_adhoc.erl b/src/mod_adhoc.erl
index 73c7d82af..6cbc99aec 100644
--- a/src/mod_adhoc.erl
+++ b/src/mod_adhoc.erl
@@ -39,7 +39,7 @@
 	 mod_options/1, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 start(Host, _Opts) ->
diff --git a/src/mod_admin_extra.erl b/src/mod_admin_extra.erl
index 9834acf01..e199f7d30 100644
--- a/src/mod_admin_extra.erl
+++ b/src/mod_admin_extra.erl
@@ -85,7 +85,7 @@
 -include("mod_roster.hrl").
 -include("mod_privacy.hrl").
 -include("ejabberd_sm.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 %%%
 %%% gen_mod
diff --git a/src/mod_admin_update_sql.erl b/src/mod_admin_update_sql.erl
index 02beb4bf8..9225d2bf0 100644
--- a/src/mod_admin_update_sql.erl
+++ b/src/mod_admin_update_sql.erl
@@ -37,7 +37,7 @@
 
 -include("logger.hrl").
 -include("ejabberd_commands.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_sql_pt.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_announce.erl b/src/mod_announce.erl
index 21213615c..d5fb38596 100644
--- a/src/mod_announce.erl
+++ b/src/mod_announce.erl
@@ -51,7 +51,7 @@
 	 announce_all_hosts_motd_delete/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_announce.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_announce_mnesia.erl b/src/mod_announce_mnesia.erl
index 068406e01..074bebc90 100644
--- a/src/mod_announce_mnesia.erl
+++ b/src/mod_announce_mnesia.erl
@@ -31,7 +31,7 @@
 	 get_motd/1, is_motd_user/2, set_motd_user/2, import/3]).
 -export([need_transform/1, transform/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_announce.hrl").
 -include("logger.hrl").
 
diff --git a/src/mod_announce_sql.erl b/src/mod_announce_sql.erl
index bc79fc77a..4c3052f16 100644
--- a/src/mod_announce_sql.erl
+++ b/src/mod_announce_sql.erl
@@ -32,7 +32,7 @@
 	 get_motd/1, is_motd_user/2, set_motd_user/2, import/3,
 	 export/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_announce.hrl").
 -include("ejabberd_sql_pt.hrl").
 -include("logger.hrl").
diff --git a/src/mod_avatar.erl b/src/mod_avatar.erl
index 09329853d..9f5e54e22 100644
--- a/src/mod_avatar.erl
+++ b/src/mod_avatar.erl
@@ -31,7 +31,7 @@
 -export([pubsub_publish_item/6, vcard_iq_convert/1, vcard_iq_publish/1,
 	 get_sm_features/5]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("pubsub.hrl").
 -include("translate.hrl").
diff --git a/src/mod_block_strangers.erl b/src/mod_block_strangers.erl
index 92f52236d..59cd8cce5 100644
--- a/src/mod_block_strangers.erl
+++ b/src/mod_block_strangers.erl
@@ -34,7 +34,7 @@
 
 -export([filter_packet/1, filter_offline_msg/1, filter_subscription/2]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_blocking.erl b/src/mod_blocking.erl
index 7d4b1bff5..853d40fe4 100644
--- a/src/mod_blocking.erl
+++ b/src/mod_blocking.erl
@@ -33,7 +33,7 @@
 	 disco_features/5, mod_options/1, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_privacy.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_bosh.erl b/src/mod_bosh.erl
index 37a7fa809..ce57faac2 100644
--- a/src/mod_bosh.erl
+++ b/src/mod_bosh.erl
@@ -40,7 +40,7 @@
 
 -include("logger.hrl").
 -include_lib("stdlib/include/ms_transform.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_http.hrl").
 -include("bosh.hrl").
 -include("translate.hrl").
diff --git a/src/mod_caps.erl b/src/mod_caps.erl
index bc48dac6f..fd3b3a1ad 100644
--- a/src/mod_caps.erl
+++ b/src/mod_caps.erl
@@ -54,7 +54,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_caps.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_carboncopy.erl b/src/mod_carboncopy.erl
index a4e3a7999..e53b573c8 100644
--- a/src/mod_carboncopy.erl
+++ b/src/mod_carboncopy.erl
@@ -42,7 +42,7 @@
 -export([list/2]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -type direction() :: sent | received.
diff --git a/src/mod_client_state.erl b/src/mod_client_state.erl
index 3585ec3a4..5ac4ad845 100644
--- a/src/mod_client_state.erl
+++ b/src/mod_client_state.erl
@@ -42,7 +42,7 @@
 	 c2s_copy_session/2, c2s_session_resumed/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -define(CSI_QUEUE_MAX, 100).
diff --git a/src/mod_configure.erl b/src/mod_configure.erl
index 32c7ebb31..b6ab8cdfe 100644
--- a/src/mod_configure.erl
+++ b/src/mod_configure.erl
@@ -39,7 +39,7 @@
 	 depends/2, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_sm.hrl").
 -include("translate.hrl").
 -include_lib("stdlib/include/ms_transform.hrl").
diff --git a/src/mod_conversejs.erl b/src/mod_conversejs.erl
index 8683d60ab..2d3dfb63b 100644
--- a/src/mod_conversejs.erl
+++ b/src/mod_conversejs.erl
@@ -32,7 +32,7 @@
 -export([start/2, stop/1, reload/3, process/2, depends/2,
          mod_opt_type/1, mod_options/1, mod_doc/0]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("ejabberd_http.hrl").
 -include("translate.hrl").
diff --git a/src/mod_delegation.erl b/src/mod_delegation.erl
index 3ae3b8a51..93c1e78d6 100644
--- a/src/mod_delegation.erl
+++ b/src/mod_delegation.erl
@@ -42,7 +42,7 @@
 	 disco_local_identity/5, disco_sm_identity/5]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -type route_type() :: ejabberd_sm | ejabberd_local.
diff --git a/src/mod_disco.erl b/src/mod_disco.erl
index deb9d15a2..c8788a77c 100644
--- a/src/mod_disco.erl
+++ b/src/mod_disco.erl
@@ -42,7 +42,7 @@
 
 -include("logger.hrl").
 -include("translate.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include_lib("stdlib/include/ms_transform.hrl").
 -include("mod_roster.hrl").
 
diff --git a/src/mod_fail2ban.erl b/src/mod_fail2ban.erl
index 8751653d8..901f71fc2 100644
--- a/src/mod_fail2ban.erl
+++ b/src/mod_fail2ban.erl
@@ -42,7 +42,7 @@
 -include_lib("stdlib/include/ms_transform.hrl").
 -include("ejabberd_commands.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -define(CLEAN_INTERVAL, timer:minutes(10)).
diff --git a/src/mod_http_api.erl b/src/mod_http_api.erl
index 427833584..6ee1fcb72 100644
--- a/src/mod_http_api.erl
+++ b/src/mod_http_api.erl
@@ -32,7 +32,7 @@
 -export([start/2, stop/1, reload/3, process/2, depends/2,
 	 mod_options/1, mod_doc/0]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("ejabberd_http.hrl").
 -include("ejabberd_stacktrace.hrl").
diff --git a/src/mod_http_upload.erl b/src/mod_http_upload.erl
index 1ff94eb4a..ce1a667e2 100644
--- a/src/mod_http_upload.erl
+++ b/src/mod_http_upload.erl
@@ -87,7 +87,7 @@
 	 expand_host/2]).
 
 -include("ejabberd_http.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_http_upload_quota.erl b/src/mod_http_upload_quota.erl
index 5ed7fcefb..4fc591ca5 100644
--- a/src/mod_http_upload_quota.erl
+++ b/src/mod_http_upload_quota.erl
@@ -51,7 +51,7 @@
 %% ejabberd_hooks callback.
 -export([handle_slot_request/6]).
 
--include_lib("xmpp/include/jid.hrl").
+-include_lib("p1_xmpp/include/jid.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
 -include_lib("kernel/include/file.hrl").
diff --git a/src/mod_jidprep.erl b/src/mod_jidprep.erl
index c666f8544..b533a2f4f 100644
--- a/src/mod_jidprep.erl
+++ b/src/mod_jidprep.erl
@@ -41,7 +41,7 @@
 
 -include("logger.hrl").
 -include("translate.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 %%--------------------------------------------------------------------
 %% gen_mod callbacks.
diff --git a/src/mod_last.erl b/src/mod_last.erl
index a7d36c791..649a94698 100644
--- a/src/mod_last.erl
+++ b/src/mod_last.erl
@@ -38,7 +38,7 @@
 	 register_user/2, depends/2, privacy_check_packet/4]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_privacy.hrl").
 -include("mod_last.hrl").
 -include("translate.hrl").
diff --git a/src/mod_legacy_auth.erl b/src/mod_legacy_auth.erl
index 2b29b4cd2..ba13a24a8 100644
--- a/src/mod_legacy_auth.erl
+++ b/src/mod_legacy_auth.erl
@@ -29,7 +29,7 @@
 %% hooks
 -export([c2s_unauthenticated_packet/2, c2s_stream_features/2]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -type c2s_state() :: ejabberd_c2s:state().
diff --git a/src/mod_mam.erl b/src/mod_mam.erl
index 9bf154f58..444655727 100644
--- a/src/mod_mam.erl
+++ b/src/mod_mam.erl
@@ -45,7 +45,7 @@
 	 is_empty_for_user/2, is_empty_for_room/3, check_create_room/4,
 	 process_iq/3, store_mam_message/7, make_id/0, wrap_as_mucsub/2, select/7]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("mod_muc_room.hrl").
 -include("ejabberd_commands.hrl").
diff --git a/src/mod_mam_mnesia.erl b/src/mod_mam_mnesia.erl
index d21d0e6ce..9dfc6d9e0 100644
--- a/src/mod_mam_mnesia.erl
+++ b/src/mod_mam_mnesia.erl
@@ -32,7 +32,7 @@
 	 is_empty_for_user/2, is_empty_for_room/3]).
 
 -include_lib("stdlib/include/ms_transform.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("mod_mam.hrl").
 
diff --git a/src/mod_mam_sql.erl b/src/mod_mam_sql.erl
index 269b4c963..b81da5b0c 100644
--- a/src/mod_mam_sql.erl
+++ b/src/mod_mam_sql.erl
@@ -33,7 +33,7 @@
 	 is_empty_for_user/2, is_empty_for_room/3, select_with_mucsub/6]).
 
 -include_lib("stdlib/include/ms_transform.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_mam.hrl").
 -include("logger.hrl").
 -include("ejabberd_sql_pt.hrl").
diff --git a/src/mod_metrics.erl b/src/mod_metrics.erl
index d42f79112..de32aebef 100644
--- a/src/mod_metrics.erl
+++ b/src/mod_metrics.erl
@@ -29,7 +29,7 @@
 -behaviour(gen_mod).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -export([start/2, stop/1, mod_opt_type/1, mod_options/1, depends/2, reload/3]).
diff --git a/src/mod_mix.erl b/src/mod_mix.erl
index 002ef5696..57de16a7a 100644
--- a/src/mod_mix.erl
+++ b/src/mod_mix.erl
@@ -41,7 +41,7 @@
 	 process_mam_query/1,
 	 process_pubsub_query/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
 -include("ejabberd_stacktrace.hrl").
diff --git a/src/mod_mix_pam.erl b/src/mod_mix_pam.erl
index 1fa5c1861..e27578028 100644
--- a/src/mod_mix_pam.erl
+++ b/src/mod_mix_pam.erl
@@ -33,7 +33,7 @@
 	 remove_user/2,
 	 process_iq/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_mqtt_session.erl b/src/mod_mqtt_session.erl
index ca025e3d2..a541e1fb0 100644
--- a/src/mod_mqtt_session.erl
+++ b/src/mod_mqtt_session.erl
@@ -28,7 +28,7 @@
 
 -include("logger.hrl").
 -include("mqtt.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -record(state, {vsn = ?VSN            :: integer(),
                 version               :: undefined | mqtt_version(),
diff --git a/src/mod_mqtt_ws.erl b/src/mod_mqtt_ws.erl
index 01f182d30..b0a2ecdbd 100644
--- a/src/mod_mqtt_ws.erl
+++ b/src/mod_mqtt_ws.erl
@@ -29,7 +29,7 @@
 -export([init/1, handle_call/3, handle_cast/2, handle_info/2,
 	 terminate/2, code_change/3, format_status/2]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_http.hrl").
 -include("logger.hrl").
 
diff --git a/src/mod_muc.erl b/src/mod_muc.erl
index b2ebc5c61..59c856466 100644
--- a/src/mod_muc.erl
+++ b/src/mod_muc.erl
@@ -77,7 +77,7 @@
 	 mod_opt_type/1, mod_options/1, depends/2]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_muc.hrl").
 -include("mod_muc_room.hrl").
 -include("translate.hrl").
diff --git a/src/mod_muc_admin.erl b/src/mod_muc_admin.erl
index ce4665d7e..ef3f5dcc3 100644
--- a/src/mod_muc_admin.erl
+++ b/src/mod_muc_admin.erl
@@ -44,7 +44,7 @@
 	 web_page_host/3, mod_options/1, get_commands_spec/0, find_hosts/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_muc.hrl").
 -include("mod_muc_room.hrl").
 -include("ejabberd_http.hrl").
diff --git a/src/mod_muc_log.erl b/src/mod_muc_log.erl
index 8bcbc8bc0..36d9a5206 100644
--- a/src/mod_muc_log.erl
+++ b/src/mod_muc_log.erl
@@ -42,7 +42,7 @@
 	 mod_opt_type/1, mod_options/1, depends/2, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_muc_room.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_muc_mnesia.erl b/src/mod_muc_mnesia.erl
index c822acc68..3847add23 100644
--- a/src/mod_muc_mnesia.erl
+++ b/src/mod_muc_mnesia.erl
@@ -44,7 +44,7 @@
 
 -include("mod_muc.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include_lib("stdlib/include/ms_transform.hrl").
 
 -record(state, {}).
diff --git a/src/mod_muc_room.erl b/src/mod_muc_room.erl
index e8b0d1bce..7b1f5ff4c 100644
--- a/src/mod_muc_room.erl
+++ b/src/mod_muc_room.erl
@@ -68,7 +68,7 @@
 	 code_change/4]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 -include("mod_muc_room.hrl").
 -include("ejabberd_stacktrace.hrl").
diff --git a/src/mod_muc_sql.erl b/src/mod_muc_sql.erl
index 1310cde7b..2c8b87d7e 100644
--- a/src/mod_muc_sql.erl
+++ b/src/mod_muc_sql.erl
@@ -42,7 +42,7 @@
 -export([set_affiliation/6, set_affiliations/4, get_affiliation/5,
 	 get_affiliations/3, search_affiliation/4]).
 
--include_lib("xmpp/include/jid.hrl").
+-include_lib("p1_xmpp/include/jid.hrl").
 -include("mod_muc.hrl").
 -include("logger.hrl").
 -include("ejabberd_sql_pt.hrl").
diff --git a/src/mod_multicast.erl b/src/mod_multicast.erl
index fa076da70..b9ea7ce32 100644
--- a/src/mod_multicast.erl
+++ b/src/mod_multicast.erl
@@ -45,7 +45,7 @@
 
 -include("logger.hrl").
 -include("translate.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -record(multicastc, {rserver :: binary(),
 		     response,
diff --git a/src/mod_offline.erl b/src/mod_offline.erl
index 1d367eb72..2a7a623e2 100644
--- a/src/mod_offline.erl
+++ b/src/mod_offline.erl
@@ -69,7 +69,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -include("ejabberd_http.hrl").
 
diff --git a/src/mod_offline_mnesia.erl b/src/mod_offline_mnesia.erl
index a3c902fb3..4cb328d2a 100644
--- a/src/mod_offline_mnesia.erl
+++ b/src/mod_offline_mnesia.erl
@@ -32,7 +32,7 @@
 	 remove_all_messages/2, count_messages/2, import/1]).
 -export([need_transform/1, transform/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_offline.hrl").
 -include("logger.hrl").
 
diff --git a/src/mod_offline_sql.erl b/src/mod_offline_sql.erl
index 4e8441b7f..9d4170725 100644
--- a/src/mod_offline_sql.erl
+++ b/src/mod_offline_sql.erl
@@ -32,7 +32,7 @@
 	 read_message/3, remove_message/3, read_all_messages/2,
 	 remove_all_messages/2, count_messages/2, import/1, export/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_offline.hrl").
 -include("logger.hrl").
 -include("ejabberd_sql_pt.hrl").
diff --git a/src/mod_ping.erl b/src/mod_ping.erl
index f233b2ae8..b528040d1 100644
--- a/src/mod_ping.erl
+++ b/src/mod_ping.erl
@@ -35,7 +35,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -include("translate.hrl").
 
diff --git a/src/mod_pres_counter.erl b/src/mod_pres_counter.erl
index 77f42961f..066e5ce4f 100644
--- a/src/mod_pres_counter.erl
+++ b/src/mod_pres_counter.erl
@@ -32,7 +32,7 @@
 
 -include("logger.hrl").
 -include("translate.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -record(pres_counter,
 	{dir, start, count, logged = false}).
diff --git a/src/mod_privacy.erl b/src/mod_privacy.erl
index 5ac26c2f5..e8971af65 100644
--- a/src/mod_privacy.erl
+++ b/src/mod_privacy.erl
@@ -41,7 +41,7 @@
 	 mod_opt_type/1, mod_options/1, depends/2]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_privacy.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_privacy_mnesia.erl b/src/mod_privacy_mnesia.erl
index 2f9cb0027..d90f30eb8 100644
--- a/src/mod_privacy_mnesia.erl
+++ b/src/mod_privacy_mnesia.erl
@@ -32,7 +32,7 @@
 	 remove_list/3, use_cache/1, import/1]).
 -export([need_transform/1, transform/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_privacy.hrl").
 -include("logger.hrl").
 
diff --git a/src/mod_privacy_sql.erl b/src/mod_privacy_sql.erl
index b27442724..234e59061 100644
--- a/src/mod_privacy_sql.erl
+++ b/src/mod_privacy_sql.erl
@@ -34,7 +34,7 @@
 
 -export([item_to_raw/1, raw_to_item/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_privacy.hrl").
 -include("logger.hrl").
 -include("ejabberd_sql_pt.hrl").
diff --git a/src/mod_private.erl b/src/mod_private.erl
index 436aae222..56bfd4b07 100644
--- a/src/mod_private.erl
+++ b/src/mod_private.erl
@@ -40,7 +40,7 @@
 -export([get_commands_spec/0, bookmarks_to_pep/2]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_private.hrl").
 -include("ejabberd_commands.hrl").
 -include("translate.hrl").
diff --git a/src/mod_private_mnesia.erl b/src/mod_private_mnesia.erl
index b5617d96e..fd4ba657b 100644
--- a/src/mod_private_mnesia.erl
+++ b/src/mod_private_mnesia.erl
@@ -31,7 +31,7 @@
 	 use_cache/1, import/3]).
 -export([need_transform/1, transform/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_private.hrl").
 -include("logger.hrl").
 
diff --git a/src/mod_private_sql.erl b/src/mod_private_sql.erl
index 08a13e8a4..511cd5c63 100644
--- a/src/mod_private_sql.erl
+++ b/src/mod_private_sql.erl
@@ -29,7 +29,7 @@
 -export([init/2, set_data/3, get_data/3, get_all_data/2, del_data/2,
 	 import/3, export/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_private.hrl").
 -include("ejabberd_sql_pt.hrl").
 -include("logger.hrl").
diff --git a/src/mod_privilege.erl b/src/mod_privilege.erl
index 353a8da27..50be843ac 100644
--- a/src/mod_privilege.erl
+++ b/src/mod_privilege.erl
@@ -41,7 +41,7 @@
 	 process_presence_out/1, process_presence_in/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -type roster_permission() :: both | get | set.
diff --git a/src/mod_proxy65_service.erl b/src/mod_proxy65_service.erl
index 78a8d85d2..98d9987db 100644
--- a/src/mod_proxy65_service.erl
+++ b/src/mod_proxy65_service.erl
@@ -38,7 +38,7 @@
 	 delete_listener/1, route/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 -include("ejabberd_stacktrace.hrl").
 
diff --git a/src/mod_pubsub.erl b/src/mod_pubsub.erl
index 76092f1c6..0b55cd7fc 100644
--- a/src/mod_pubsub.erl
+++ b/src/mod_pubsub.erl
@@ -40,7 +40,7 @@
 -protocol({xep, 248, '0.2'}).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("pubsub.hrl").
 -include("mod_roster.hrl").
 -include("translate.hrl").
diff --git a/src/mod_push.erl b/src/mod_push.erl
index 5477c5792..eab812226 100644
--- a/src/mod_push.erl
+++ b/src/mod_push.erl
@@ -51,7 +51,7 @@
 
 -include("ejabberd_commands.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -define(PUSH_CACHE, push_cache).
diff --git a/src/mod_push_keepalive.erl b/src/mod_push_keepalive.erl
index e0e83f1e1..c57f2404a 100644
--- a/src/mod_push_keepalive.erl
+++ b/src/mod_push_keepalive.erl
@@ -36,7 +36,7 @@
 	 c2s_handle_cast/2, c2s_handle_info/2, c2s_stanza/3]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -define(PUSH_BEFORE_TIMEOUT_PERIOD, 120000). % 2 minutes.
diff --git a/src/mod_push_mnesia.erl b/src/mod_push_mnesia.erl
index 665ab48f2..0e17d11f2 100644
--- a/src/mod_push_mnesia.erl
+++ b/src/mod_push_mnesia.erl
@@ -35,7 +35,7 @@
 
 -include_lib("stdlib/include/ms_transform.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_push.hrl").
 
 %%%-------------------------------------------------------------------
diff --git a/src/mod_push_sql.erl b/src/mod_push_sql.erl
index c024a12d1..a87f33f47 100644
--- a/src/mod_push_sql.erl
+++ b/src/mod_push_sql.erl
@@ -31,7 +31,7 @@
 	 lookup_sessions/3, lookup_sessions/2, lookup_sessions/1,
 	 delete_session/3, delete_old_sessions/2, export/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("ejabberd_sql_pt.hrl").
 -include("mod_push.hrl").
diff --git a/src/mod_register.erl b/src/mod_register.erl
index b85efd57c..890852821 100644
--- a/src/mod_register.erl
+++ b/src/mod_register.erl
@@ -40,7 +40,7 @@
 -deprecated({try_register, 4}).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 start(Host, _Opts) ->
diff --git a/src/mod_register_web.erl b/src/mod_register_web.erl
index 0cf4bcff8..43a123ead 100644
--- a/src/mod_register_web.erl
+++ b/src/mod_register_web.erl
@@ -34,7 +34,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -include("ejabberd_http.hrl").
 
diff --git a/src/mod_roster.erl b/src/mod_roster.erl
index 94cae4950..8c9206e4b 100644
--- a/src/mod_roster.erl
+++ b/src/mod_roster.erl
@@ -54,7 +54,7 @@
 	 depends/2, set_item_and_notify_clients/3]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_roster.hrl").
 -include("ejabberd_http.hrl").
 -include("ejabberd_web_admin.hrl").
diff --git a/src/mod_roster_mnesia.erl b/src/mod_roster_mnesia.erl
index fa1d52f7c..ced28ce60 100644
--- a/src/mod_roster_mnesia.erl
+++ b/src/mod_roster_mnesia.erl
@@ -37,7 +37,7 @@
 
 -include("mod_roster.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 %%%===================================================================
 %%% API
diff --git a/src/mod_roster_sql.erl b/src/mod_roster_sql.erl
index ebfcde463..3510b4624 100644
--- a/src/mod_roster_sql.erl
+++ b/src/mod_roster_sql.erl
@@ -38,7 +38,7 @@
 -include("mod_roster.hrl").
 -include("ejabberd_sql_pt.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/jid.hrl").
+-include_lib("p1_xmpp/include/jid.hrl").
 
 %%%===================================================================
 %%% API
diff --git a/src/mod_s2s_dialback.erl b/src/mod_s2s_dialback.erl
index 05747e3bc..e2bdc847e 100644
--- a/src/mod_s2s_dialback.erl
+++ b/src/mod_s2s_dialback.erl
@@ -33,7 +33,7 @@
 	 s2s_in_features/2, s2s_out_init/2, s2s_out_closed/2,
 	 s2s_out_tls_verify/2]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
 
diff --git a/src/mod_service_log.erl b/src/mod_service_log.erl
index f5f23eeb0..e8beaac9a 100644
--- a/src/mod_service_log.erl
+++ b/src/mod_service_log.erl
@@ -34,7 +34,7 @@
 
 -include("logger.hrl").
 -include("translate.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 start(Host, _Opts) ->
     ejabberd_hooks:add(user_send_packet, Host, ?MODULE,
diff --git a/src/mod_shared_roster.erl b/src/mod_shared_roster.erl
index 13ff90466..46fbbd535 100644
--- a/src/mod_shared_roster.erl
+++ b/src/mod_shared_roster.erl
@@ -43,7 +43,7 @@
 
 -include("logger.hrl").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -include("mod_roster.hrl").
 
diff --git a/src/mod_shared_roster_ldap.erl b/src/mod_shared_roster_ldap.erl
index 08fbe8793..8f6cff118 100644
--- a/src/mod_shared_roster_ldap.erl
+++ b/src/mod_shared_roster_ldap.erl
@@ -43,7 +43,7 @@
 	 depends/2, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_roster.hrl").
 -include("eldap.hrl").
 -include("translate.hrl").
diff --git a/src/mod_shared_roster_mnesia.erl b/src/mod_shared_roster_mnesia.erl
index 484df0cd5..60a45941a 100644
--- a/src/mod_shared_roster_mnesia.erl
+++ b/src/mod_shared_roster_mnesia.erl
@@ -37,7 +37,7 @@
 -include("mod_roster.hrl").
 -include("mod_shared_roster.hrl").
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 %%%===================================================================
 %%% API
diff --git a/src/mod_shared_roster_sql.erl b/src/mod_shared_roster_sql.erl
index 0d44dce69..0aacf03e8 100644
--- a/src/mod_shared_roster_sql.erl
+++ b/src/mod_shared_roster_sql.erl
@@ -35,7 +35,7 @@
 	 add_user_to_group/3, remove_user_from_group/3, import/3,
 	 export/1]).
 
--include_lib("xmpp/include/jid.hrl").
+-include_lib("p1_xmpp/include/jid.hrl").
 -include("mod_roster.hrl").
 -include("mod_shared_roster.hrl").
 -include("ejabberd_sql_pt.hrl").
diff --git a/src/mod_sic.erl b/src/mod_sic.erl
index 4eeb8fd08..2ad4ce5a6 100644
--- a/src/mod_sic.erl
+++ b/src/mod_sic.erl
@@ -35,7 +35,7 @@
 	 process_sm_iq/1, mod_options/1, depends/2, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 start(Host, _Opts) ->
diff --git a/src/mod_sip.erl b/src/mod_sip.erl
index 6ea2a408f..3bc901035 100644
--- a/src/mod_sip.erl
+++ b/src/mod_sip.erl
@@ -55,7 +55,7 @@ mod_doc() ->
 	 locate/1, mod_opt_type/1, mod_options/1, depends/2,
          mod_doc/0]).
 
--include_lib("esip/include/esip.hrl").
+-include_lib("p1_sip/include/esip.hrl").
 
 %%%===================================================================
 %%% API
diff --git a/src/mod_sip_proxy.erl b/src/mod_sip_proxy.erl
index ddfefb1ad..e1c8a734f 100644
--- a/src/mod_sip_proxy.erl
+++ b/src/mod_sip_proxy.erl
@@ -39,7 +39,7 @@
 	 code_change/4]).
 
 -include("logger.hrl").
--include_lib("esip/include/esip.hrl").
+-include_lib("p1_sip/include/esip.hrl").
 
 -define(SIGN_LIFETIME, 300). %% in seconds.
 
diff --git a/src/mod_sip_registrar.erl b/src/mod_sip_registrar.erl
index 5079aebf0..73a5a5168 100644
--- a/src/mod_sip_registrar.erl
+++ b/src/mod_sip_registrar.erl
@@ -41,7 +41,7 @@
 	 terminate/2, code_change/3]).
 
 -include("logger.hrl").
--include_lib("esip/include/esip.hrl").
+-include_lib("p1_sip/include/esip.hrl").
 
 -define(CALL_TIMEOUT, timer:seconds(30)).
 -define(DEFAULT_EXPIRES, 3600).
diff --git a/src/mod_stats.erl b/src/mod_stats.erl
index c61fe6175..b73227582 100644
--- a/src/mod_stats.erl
+++ b/src/mod_stats.erl
@@ -35,7 +35,7 @@
 	 mod_options/1, depends/2, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 start(Host, _Opts) ->
diff --git a/src/mod_stream_mgmt.erl b/src/mod_stream_mgmt.erl
index f60f6722b..b5d2def60 100644
--- a/src/mod_stream_mgmt.erl
+++ b/src/mod_stream_mgmt.erl
@@ -37,7 +37,7 @@
 %% adjust pending session timeout / access queue
 -export([get_resume_timeout/1, set_resume_timeout/2, queue_find/2]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include_lib("p1_utils/include/p1_queue.hrl").
 -include("translate.hrl").
diff --git a/src/mod_stun_disco.erl b/src/mod_stun_disco.erl
index 6e7592453..203ac735e 100644
--- a/src/mod_stun_disco.erl
+++ b/src/mod_stun_disco.erl
@@ -55,7 +55,7 @@
 
 -include("logger.hrl").
 -include("translate.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -define(STUN_MODULE, ejabberd_stun).
 
diff --git a/src/mod_time.erl b/src/mod_time.erl
index 4ec3cd64e..9b2072ad5 100644
--- a/src/mod_time.erl
+++ b/src/mod_time.erl
@@ -36,7 +36,7 @@
 	 mod_options/1, depends/2, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 start(Host, _Opts) ->
diff --git a/src/mod_vcard.erl b/src/mod_vcard.erl
index 8e0d32a4a..e4687c61e 100644
--- a/src/mod_vcard.erl
+++ b/src/mod_vcard.erl
@@ -44,7 +44,7 @@
 -export([route/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_vcard.hrl").
 -include("translate.hrl").
 -include("ejabberd_stacktrace.hrl").
diff --git a/src/mod_vcard_ldap.erl b/src/mod_vcard_ldap.erl
index bc6e7ebca..955489475 100644
--- a/src/mod_vcard_ldap.erl
+++ b/src/mod_vcard_ldap.erl
@@ -40,7 +40,7 @@
 
 -include("logger.hrl").
 -include("eldap.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -define(PROCNAME, ejabberd_mod_vcard_ldap).
diff --git a/src/mod_vcard_mnesia.erl b/src/mod_vcard_mnesia.erl
index ad4be701c..d0dafc10d 100644
--- a/src/mod_vcard_mnesia.erl
+++ b/src/mod_vcard_mnesia.erl
@@ -33,7 +33,7 @@
 -export([need_transform/1, transform/1]).
 -export([mod_opt_type/1, mod_options/1, mod_doc/0]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_vcard.hrl").
 -include("logger.hrl").
 -include("translate.hrl").
diff --git a/src/mod_vcard_sql.erl b/src/mod_vcard_sql.erl
index 620c1b4b5..8e8ea10ff 100644
--- a/src/mod_vcard_sql.erl
+++ b/src/mod_vcard_sql.erl
@@ -32,7 +32,7 @@
 	 search_fields/1, search_reported/1, import/3, export/1]).
 -export([is_search_supported/1]).
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("mod_vcard.hrl").
 -include("logger.hrl").
 -include("ejabberd_sql_pt.hrl").
diff --git a/src/mod_vcard_xupdate.erl b/src/mod_vcard_xupdate.erl
index 59ebc7f71..fd75decd4 100644
--- a/src/mod_vcard_xupdate.erl
+++ b/src/mod_vcard_xupdate.erl
@@ -37,7 +37,7 @@
 -export([compute_hash/1]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -define(VCARD_XUPDATE_CACHE, vcard_xupdate_cache).
diff --git a/src/mod_version.erl b/src/mod_version.erl
index 9aa93829d..aa89e8253 100644
--- a/src/mod_version.erl
+++ b/src/mod_version.erl
@@ -35,7 +35,7 @@
 	 mod_opt_type/1, mod_options/1, depends/2, mod_doc/0]).
 
 -include("logger.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 start(Host, _Opts) ->
diff --git a/src/node_flat.erl b/src/node_flat.erl
index 55dea0d8d..0dada8e22 100644
--- a/src/node_flat.erl
+++ b/src/node_flat.erl
@@ -34,7 +34,7 @@
 -author('christophe.romain@process-one.net').
 
 -include("pubsub.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 -export([init/3, terminate/2, options/0, features/0,
     create_node_permission/6, create_node/2, delete_node/1,
diff --git a/src/node_flat_sql.erl b/src/node_flat_sql.erl
index f9c8a209d..9d33f0376 100644
--- a/src/node_flat_sql.erl
+++ b/src/node_flat_sql.erl
@@ -35,7 +35,7 @@
 
 
 -include("pubsub.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_sql_pt.hrl").
 -include("translate.hrl").
 
diff --git a/src/nodetree_tree.erl b/src/nodetree_tree.erl
index 853c1fb93..28ad2520d 100644
--- a/src/nodetree_tree.erl
+++ b/src/nodetree_tree.erl
@@ -41,7 +41,7 @@
 -include_lib("stdlib/include/ms_transform.hrl").
 
 -include("pubsub.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -export([init/3, terminate/2, options/0, set_node/1,
diff --git a/src/nodetree_tree_sql.erl b/src/nodetree_tree_sql.erl
index 402c50901..d5f82c88c 100644
--- a/src/nodetree_tree_sql.erl
+++ b/src/nodetree_tree_sql.erl
@@ -39,7 +39,7 @@
 
 
 -include("pubsub.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("ejabberd_sql_pt.hrl").
 -include("translate.hrl").
 
diff --git a/src/prosody2ejabberd.erl b/src/prosody2ejabberd.erl
index 3992a4034..d0d186015 100644
--- a/src/prosody2ejabberd.erl
+++ b/src/prosody2ejabberd.erl
@@ -27,8 +27,8 @@
 %% API
 -export([from_dir/1]).
 
--include_lib("xmpp/include/scram.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/scram.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("logger.hrl").
 -include("mod_roster.hrl").
 -include("mod_offline.hrl").
diff --git a/src/pubsub_subscription.erl b/src/pubsub_subscription.erl
index 92d6d314d..716a29962 100644
--- a/src/pubsub_subscription.erl
+++ b/src/pubsub_subscription.erl
@@ -38,7 +38,7 @@
     read_subscription/3, write_subscription/4]).
 
 -include("pubsub.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -define(PUBSUB_DELIVER, <<"pubsub#deliver">>).
diff --git a/src/pubsub_subscription_sql.erl b/src/pubsub_subscription_sql.erl
index c36934f50..8eb6f4db1 100644
--- a/src/pubsub_subscription_sql.erl
+++ b/src/pubsub_subscription_sql.erl
@@ -34,7 +34,7 @@
     get_options_xform/2, parse_options_xform/1]).
 
 -include("pubsub.hrl").
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 -include("translate.hrl").
 
 -define(PUBSUB_DELIVER, <<"pubsub#deliver">>).
diff --git a/test/suite.hrl b/test/suite.hrl
index d8ea3e23b..9c6bc3f42 100644
--- a/test/suite.hrl
+++ b/test/suite.hrl
@@ -1,8 +1,8 @@
 -include_lib("common_test/include/ct.hrl").
--include_lib("fast_xml/include/fxml.hrl").
--include_lib("xmpp/include/jid.hrl").
--include_lib("xmpp/include/ns.hrl").
--include_lib("xmpp/include/xmpp_codec.hrl").
+-include_lib("p1_xml/include/fxml.hrl").
+-include_lib("p1_xmpp/include/jid.hrl").
+-include_lib("p1_xmpp/include/ns.hrl").
+-include_lib("p1_xmpp/include/xmpp_codec.hrl").
 -include("mod_proxy65.hrl").
 
 -define(STREAM_TRAILER, <<"</stream:stream>">>).
diff --git a/tools/xml_compress_gen.erl b/tools/xml_compress_gen.erl
index 5b43c5925..087a6c13b 100644
--- a/tools/xml_compress_gen.erl
+++ b/tools/xml_compress_gen.erl
@@ -24,7 +24,7 @@
 -module(xml_compress_gen).
 -author("pawel@process-one.net").
 
--include_lib("xmpp/include/xmpp.hrl").
+-include_lib("p1_xmpp/include/xmpp.hrl").
 
 %% API
 -export([archive_analyze/3, process_stats/1, gen_code/3]).
-- 
2.35.3

